scripts/manage_case

```diff
@@ -1,375 +1,172 @@
-    print "                Model Grids:                           \n";
-    print " ======================================================\n";
-    print "\n";
-    foreach my $node ($xml_grid->findnodes(".//grids/grid")) {
-	my @lname_nodes = $node->findnodes("./lname");
-	my $lname = $lname_nodes[0]->textContent();
-	my @alias_nodes = $node->findnodes("./alias");
-	my $alias;
-	if ($#alias_nodes == 0) {
-	    $alias = $alias_nodes[0]->textContent();
-	}
-	my $compset   = $node->getAttribute('compset');
-	my $support   = $node->getAttribute('support');
-
-	print " model grid: $lname \n";
-	if (defined ($alias)  ) { print "    alias        : $alias \n"}
-	if (defined ($compset)) { print "    compset match: $compset \n"}
-	if (defined ($support)) { print "    support level: $support \n";}
-	print " \n";
-    }    
-    print " \n";
-}
-#-------------------------------------------------------------------------------
-sub _queryMachines
-{
-    # Print all currently supported valid grids
-    my ($model, $cimeroot, $xml_config) = @_;
-
-    my @files = $xml_config->findnodes(".//entry[\@id=\"MACHINES_SPEC_FILE\"]/default_value");
-    if ($#files == -1) {
-	die " ERROR: in parsing $file_config for the MACHIENS_SPEC_FILE node \n";
-    } 
-    my $file = $files[0]->textContent();
-    $file =~ s/\$CIMEROOT/$cimeroot/;
-    $file =~ s/\$MODEL/$model/;
-    (-f "$file")  or  die "** Cannot find machines specification file $file ***\n";
-
-    my $model = uc $model;
-    print "\n";
-    print " ======================================================\n";
-    print "            $model Machines:                          \n";
-    print " ======================================================\n";
-    print "\n";
-
-    my $xml_machine = XML::LibXML->new( no_blanks => 1)->parse_file($file);
-    foreach my $machine ($xml_machine->findnodes(".//machine")) {
-	my $name = $machine->getAttribute('MACH');
-	foreach my $node ($machine->findnodes("./DESC")) {
-	    my $desc = $node->textContent();
-	    print " $name :  \n      $desc \n";
-	}
-    }
-}
+#!/usr/bin/env python2.7
+#FIXME - the above needs the explicit python2.7 - this is not general ???
+
+from Tools.standard_script_setup import *
+
+from CIME.utils         import expect, get_model
+from CIME.XML.files     import Files
+from CIME.XML.component import Component
+from CIME.XML.compsets  import Compsets
+from CIME.XML.grids     import Grids
+from CIME.XML.machines  import Machines
+
+import argparse, doctest
+
+###############################################################################
+def query_grids():  
+###############################################################################
+
+    files = Files()
+    config_file = files.get_value("GRIDS_SPEC_FILE", resolved=True)
+    expect(os.path.isfile(config_file),
+           "Cannot find config_file %s on disk" %config_file)
+
+    parsed_xml = Grids(config_file)
+    if verbose:
```

Comment:
You should probably use the logger stuff that Jim set up. Also, where does verbose come from in this function? I don't see it as an argument or stack variable. You may want to run pylint on this code to find mistakes like this.